cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.strimzi.api.kafka.model.KafkaRebalance;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.api.kafka.operator.assembly.KafkaRebalanceState;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class AbstractResourceStateMatchers {
+
+
+    public static Matcher<KafkaRebalance> hasState() {
+        return new TypeSafeDiagnosingMatcher<>() {
+
+            @Override
+            protected boolean matchesSafely(KafkaRebalance kafkaRebalance, Description mismatchDescription) {
+                mismatchDescription.appendText("was ").appendValue(kafkaRebalance);
+
+                if (kafkaRebalance == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status is null");
+                    return false;
+                }
+
+                if (kafkaRebalance.getStatus().getConditions() == null) {
+                    mismatchDescription.appendText("\n KafkaRebalance status conditions is null");
+                    return false;
+                }
+
+                return true;
+            }
+
+            @Override
+            public void describeTo(final Description description) {
+                //
+            }
+        };
+
+    }
+
+
+    /**
+     * Checks all conditions in the supplied resource to see if the type of one of them matches the supplied rebalance state.
+     * @param state he expected rebalance state to be searched for.
+     * @return
+     */
+    public static Matcher<List<Condition>> hasStateInConditions(KafkaRebalanceState state) {
+        return new TypeSafeDiagnosingMatcher<>() {
+
+            @Override
+            protected boolean matchesSafely(List<Condition> conditions, Description mismatchDescription) {
+                mismatchDescription.appendText("was ").appendValue(conditions);
+
+                List<String> foundStatuses = new ArrayList<>();
+
+                for (Condition condition :  conditions) {
```

Comment:
I think we need the null check in both places. the hasState method as well as the hasStateInConditions,
```suggestion
                if(conditions == null) {
                  mismatchDescription.appendText("\n Conditions empty");
                  return false;
                }
                
                for (Condition condition :  conditions) {
```